### PR TITLE
Improve the layout code samples

### LINKED ISSUE #3976: Improvements to PyQGIS cookbook - Map Rendering and Printing
## Description

Improvements to the Map Rendering and Printing Section of the PyQGIS Cookbook based on some comments from @nyalldawson in here
https://github.com/qgis/QGIS-Documentation/pull/3974#issuecomment-509852898

- layout.initializeDefaults() # this initializes the layout with some default settings, specifically by adding an empty A4 page to the layout. You can create layouts without calling this method, but you'll need to take care of adding pages to the layout yourself
- "this item tells the libraries where to put the map itself. Here we create a map and stretch it over the whole paper size" - the code doesn't do the stretch part. I think it'll create a 0 width/0 height item by default!. You probably want something like map.attemptMove(QgsLayoutPoint(0,0, QgsUnitTypes.LayoutMillimeters)) and map.attemptResize(QgsLayoutSize(200,200, QgsUnitTypes.LayoutMillimeters)). Note that these methods are called "attempt..." because certain layout or item settings (e.g. data defined position or size) will override the call - so you are asking nicely, but aren't necessarily guaranteed that the item WILL move or resize to what you've asked!
-  layout.addItem( ... ) - this should be layout.addLayoutItem( ... )

Page URL:

https://docs.qgis.org/testing/en/docs/pyqgis_developer_cookbook/composer.html#map-rendering-and-printing

### PR DESCRIPTION
**Pull request by DelazJ**
with Nyall's [suggestions](https://github.com/qgis/QGIS-Documentation/pull/3974#issuecomment-509852898) (fixes #3976)

### CODE REVIEW

File: docs/pyqgis_developer_cookbook/composer.rst [hunk 1]
```diff
@@ -137,22 +137,31 @@ class. Let us create an instance of it:
 
 .. testcode:: composer
 
-  project = QgsProject()
+  project = QgsProject.instance()
   layout = QgsPrintLayout(project)
   layout.initializeDefaults()
```
Comment (DelazJ) [inline docs/pyqgis_developer_cookbook/composer.rst:142]: Should this display a new layout in the project? I can't see one added running these commands (can't find it from the manager), but it seems to be there. Just invisible.

File: docs/pyqgis_developer_cookbook/composer.rst [hunk 2]
```diff
@@ -137,22 +137,31 @@ class. Let us create an instance of it:
...
-* map --- this item tells the libraries where to put the map itself. Here we
-  create a map and stretch it over the whole paper size
+* map --- Here we create a map of a custom size and render the current map canvas
 
   .. testcode:: composer
 
     map = QgsLayoutItemMap(layout)
-    layout.addItem(map)
+    # Set map item position and size (by default, it is a 0 width/0 height item placed at 0,0)
+    map.attemptMove(QgsLayoutPoint(5,5, QgsUnitTypes.LayoutMillimeters))
```
Comment (DelazJ) [inline docs/pyqgis_developer_cookbook/composer.rst:160]: If I use 0,0, everything is OK but using a custom point, I get a blank map item rendering (nan scale)

**Review by DelazJ — COMMENTED** [review]
@nyalldawson may I have your review, please? Thanks.

File: docs/pyqgis_developer_cookbook/composer.rst (hunk 1 above)
Comment (nyalldawson) [reply docs/pyqgis_developer_cookbook/composer.rst:142]: No, it just creates a "temporary" one. To add it to the project you'd use:

    project.layoutManager().addLayout(layout)

then it'll become visible in the layout manager and will be saved/restored along with the project.

Depending on the use case the temporary layout may be preferable, e.g. if it's just being used to export something and it should never be exposed to the user or change the project.

**Review by nyalldawson — APPROVED** [review]
Looks good!

File: docs/pyqgis_developer_cookbook/composer.rst (hunk 1 above)
Comment (DelazJ) [reply docs/pyqgis_developer_cookbook/composer.rst:142]: Oh! good to know, I'll add that information too. Thanks.

File: docs/pyqgis_developer_cookbook/composer.rst (hunk 2 above)
Comment (DelazJ) [reply docs/pyqgis_developer_cookbook/composer.rst:160]: Any idea what happens here? A bug in QGIS that prevents to display contents of a map added at a location other than (0,0)?

**Comment by DelazJ** [comment]
For reference I opened an issue report at https://github.com/qgis/QGIS/issues/40877

**Pull request merged**